website/docs/server-setup.md

```diff
@@ -49,7 +49,8 @@ const http = require('http');
 const express = require('express');
 const { graphqlHTTP } = require('express-graphql');
 const { execute, subscribe } = require('graphql');
-const { createServer } = require('graphql-ws');
+const ws = require('ws'); // yarn add ws
+const { useServer } = require('graphql-ws/lib/use/ws');
 
 const typeDefs = require('./graphql/types');
 const resolvers = require('./graphql/resolvers');
```

Comment:
@enisdenjo - can you provide an example of what `'./graphql/types'` and `'./graphql/resolvers'` look like here, please? I'm trying to setup a subscriptions demo for GraphQL Tools and cannot for the life of me get your code here to do anything meaningful. I must be missing something.

Reply:
Hey hey, I've actually refactored an existing `subscriptions-transport-ws` example and merely adjusted it to `graphql-ws`. You can check that PR over at #2282. So, I cannot give you an answer either. 😅

Maybe @ardatan can help?

Reply:
Thanks for the reply! I have now gotten it all working with `makeExecutableSchema`, which is a good start. Now starting to work through some disconnects that seem to have appeared within Schema Stitching supporting a `subscriber` function.

That said, subscriber docs in the [Remote Schemas](https://www.graphql-tools.com/docs/remote-schemas#creating-a-subscriber) section appear to also be somewhat outdated. Does this look up-to-date with `graphql-ws`...? https://github.com/gmac/schema-stitching-demos/blob/subscriptions/subscriptions/lib/make_remote_subscriber.js

Reply:
Speaking of which, it probably wouldn't hurt to mention that `webSocketImpl` param in the `graphql-ws` readme. All the examples seem to assume there's a default WebSocket implementation

Reply:
Yes, that looks just fine! I also left a comment over at https://github.com/gmac/schema-stitching-demos/pull/2. We can continue the discussion there too.

Reply:
@enisdenjo why is the unsubscribe method in the example in docs and used here calling client.subscribe?

Is that strange or just strange to someone who knows nothing about observables and async iterators but wish they did...

Reply:
Its because `graphql-ws` is ment to be _plugged_ into an Observable implementation. [Straight from the Observable recipe:](https://github.com/enisdenjo/graphql-ws#observable)
```ts
import { Observable } from 'relay-runtime';
// or
import { Observable } from '@apollo/client';
// or
import { Observable } from 'rxjs';
// or
import Observable from 'zen-observable';
// or any other lib which implements Observables as per the ECMAScript proposal: https://github.com/tc39/proposal-observable

const client = createClient({
  url: 'wss://graphql.loves/observables',
});

function toObservable(operation) {
  return new Observable((observer) => client.subscribe(operation, observer));
}

const observable = toObservable({ query: `subscription { ping }` });

const subscription = observable.subscribe({
  next: (data) => {
    expect(data).toBe({ data: { ping: 'pong' } });
  },
});

// ⏱

subscription.unsubscribe();
```

Reply:
Like I said, I am a complete newbie, but I can't see how that is equivalent to https://github.com/gmac/schema-stitching-demos/blob/subscriptions/subscriptions/lib/make_remote_subscriber.js#L12

Reply:
[The `observableToAsyncIterable` function takes in an Observable.](https://github.com/ardatan/graphql-tools/blob/b2cda0a1b81e0d50bb746de29d9e1d4957d5421e/packages/utils/src/observableToAsyncIterable.ts#L17) An observable looks like this ([from the same file](https://github.com/ardatan/graphql-tools/blob/b2cda0a1b81e0d50bb746de29d9e1d4957d5421e/packages/utils/src/observableToAsyncIterable.ts#L7-L13)):
```ts
export interface Observable<T> {
  subscribe(
    observer: Observer<T>
  ): {
    unsubscribe: () => void;
  };
}
```

Since `graphql-ws` client's subscribe function is actually a subscriber (the implementing/constructor part of an Observable) which is used to _create_ an observable, it merely returns a cleanup function (as an observable subscriber should).

So, to pass it along and fit the observable signature, without actually using one, you do this:
```ts
function likeObservable(operation): Observable {
  return {
    subscribe: (observer) => ({
      unsubscribe: () => client.subscribe(operation, observer),
    }),
  };
}
```

Reply:
I guess I understand that they subscribe or returns a cleanup function I just don't understand where the subscription is kicked off before it is cleaned up.....

As I said, I'm not familiar at all with observables. I will just have to trust that this code is okay. I think it might be beneficial for our canonical example to not use observables and to just use a sync iterators.... But I will defer to @ardatan  and you and whomever, as I am way out of my depth here

Reply:
TBH, I am a bit lost now... I guess you were referring to https://github.com/gmac/schema-stitching-demos/pull/2? I wasn't debugging, just explaining the client usage (this example is actually about the server though).

Nevertheless, if it indeed about that PR - @gmac had had some findings; lets continue the discussion there.

Reply:
I've swapped this out for the `AsyncIterator` recipe from the graphql-ws readme. Works great! https://github.com/gmac/schema-stitching-demos/pull/2/files#diff-ef742764e87393d03266fea0a8436b3799d200e36f07637b3ae912595be527b2